### PR TITLE
fix SchemaCompiler filterPrimaryKey

### PR DESCRIPTION
**Pull request by MonaMayrhofer**
I think this fixes the issue I described in #1914.

I would love to add unit tests for this, however I wasn't able to find the right place to put them - If unit tests are wanted here I'd love a little guidance.

### CODE REVIEW

**Comment by amitaibu** [comment]
> I would love to add unit tests for this

Maybe here? https://github.com/digitallyinduced/ihp/blob/master/Test/IDE/SchemaDesigner/CompilerSpec.hs

**Comment by MonaMayrhofer** [comment]
Im not sure a file in `test/IDE` is the right place tho... the code in question isn't part of the IDE per se, and all the other tests in the file are testing the compiler for Schema.sql and as far as i can tell have nothing to do with the Query Builder, right?

**Comment by amitaibu** [comment]
You are right. Let's get Marc's feedback on that :smile:

**Comment by mpscholten** [comment]
Thanks for the issue and the PR :) (and sorry for the delay in the review)

We could put it into `Test.SchemaCompilerSpec`?

**Comment by MonaMayrhofer** [comment]
That sounds reasonable!👍🏻

File: Test/SchemaCompilerSpec.hs
Comment (MonaMayrhofer) [inline Test/SchemaCompilerSpec.hs:511]: I would like to draw attention to this change here

In the test cases the instance I am looking for is the last one in the generated code, which resulted in an error, even tho i think it's totally fine. That this results in an error looked like a simple oversight to me so I changed it - the tests still pass... However a second opinion would be great in cause this actually had a deeper reason.

**Comment by MonaMayrhofer** [comment]
I think this is ready for review :blush:

File: Test/SchemaCompilerSpec.hs
Comment (mpscholten) [reply Test/SchemaCompilerSpec.hs:511]: Looks good to me 👍

**Review by mpscholten — APPROVED** [review]
Thanks 👍 looks great, will merge once tests have run through

**Pull request merged**
